### PR TITLE
refactoring - pipelinerunstate 

### PR DESCRIPTION
**Pull request by pritidesai**
# Changes

**(restructuring only - no updates to any of the functions)**

Creating a new file with all attributes of `pipelinerunstate` along with the test file (moved unit tests from `pipelinerunresolution_test.go` to `pipelinerunstate_test.go`)

Refactored following functions as an attributes of `pipelunerunstate`:

* getTaskRunsStatus
* getSkippedTasks
* GetPipelineConditionStatus

I have left all of the functions under the same package `resources` for simplicity. We can create a new package and move it under that package in a separate PR.

/kind cleanup

/cc @bobcatfish @jerop 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: pkg/reconciler/pipelinerun/pipelinerun.go [hunk 1]
```diff
@@ -469,7 +469,7 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1beta1.PipelineRun) err
 		return err
 	}
 
-	after := resources.GetPipelineConditionStatus(pr, pipelineState, logger, d, dfinally)
+	after := pipelineState.GetPipelineConditionStatus(pr, logger, d, dfinally)
```
Comment (jerop) [inline pkg/reconciler/pipelinerun/pipelinerun.go:472]: this is referred to as `pipelinestate` sometimes and `pipelinerunstate` at other times -- i think we should update to have a consistent name

**Review by jerop — COMMENTED** [review]
thank you @pritidesai!

i suggest we update the instances where `pipelinerunstate` is referred to as `pipelinestate`

File: pkg/reconciler/pipelinerun/pipelinerun.go (hunk 1 above)
Comment (pritidesai) [reply pkg/reconciler/pipelinerun/pipelinerun.go]: thanks @jerop I have renamed `pipelinestate` to `pipelineRunState` which is of type `PipelineRunState` in `pipelinerun.go`. Also, Have left the receiver as `state` in `pipelinerunstate.go` for simplicity.

File: pkg/reconciler/pipelinerun/resources/pipelinerunstate.go [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+	skipped := []v1beta1.SkippedTask{}
+	for _, rprt := range state {
+		if rprt.Skip(state, d) {
+			skipped = append(skipped, v1beta1.SkippedTask{Name: rprt.PipelineTask.Name})
+		}
+	}
+	return skipped
+}
+
+func (state PipelineRunState) GetTaskRunsStatus(pr *v1beta1.PipelineRun) map[string]*v1beta1.PipelineRunTaskRunStatus {
```
Comment (bobcatfish) [inline pkg/reconciler/pipelinerun/resources/pipelinerunstate.go:264]: nice - i like that none of these have pointer receivers so we don't have to worry about side effects :D

**Review by bobcatfish — COMMENTED** [review]
Looks like this is pretty much a "pure" refactoring, i.e. no logic changes at all, so looks great to me and i like the incremental step toward improvement.

/approve

> We can create a new package and move it under that package in a separate PR.

I think that's a great next step! It would be interesting to maybe even step back and take a look at what end state we'd like to get to

**Pull request merged**